### PR TITLE
Build a Docker image

### PR DESCRIPTION
**Pull request by neongreen**
Related to #18.

### CODE REVIEW

File: nix/grpc-ipv4.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
+diff --git a/src/core/lib/iomgr/tcp_server_posix.c b/src/core/lib/iomgr/tcp_server_posix.c
```
Comment (neongreen) [inline nix/grpc-ipv4.patch:1]: This should not be necessary once we migrate to newer gRPC.

Without this patch, we are getting:

```
$ docker run -e RUNTIME_SUBDIRECTORY=ratelimit -p 50051:50051 -v $(pwd)/test_integration_go/config:/srv/runtime_data/current juspay/fencer

2019-09-27T20:18:46Z, I, Loading rules from /srv/runtime_data/current/ratelimit/config
E0927 20:18:46.723346993       1 server_chttp2.c:53]         {"created":"@1569615526.723319018","description":"Only 1 addresses added out of total 2 resolved","file":"src/core/ext/transport/chttp2/server/chttp2_server.c","file_line":267,"referenced_errors":[{"created":"@1569615526.723315461","description":"OS Error","errno":97,"file":"src/core/lib/iomgr/socket_utils_common_posix.c","file_line":270,"os_error":"Address family not supported by protocol","syscall":"socket","target_address":"[::1]:50051"}]}
```

The problem is that IPv6 does not seem to be supported in Docker and gRPC treats `0.0.0.0` or `localhost` as "listen on both IPv4 and IPv6 wildcards".

**Comment by neongreen** [comment]
The last commit also switches GitLab CI to test the created Docker image instead of creating a Nix closure and copying it over.

This is good because we intend the Docker image to be the recommended way of running Fencer.

**Review by mdimjasevic — APPROVED** [review]
This looks good.

Is the Docker image then available from the Gitlab CI or is it just for build-testing purposes?

**Comment by neongreen** [comment]
We're going to push it to Dockerhub, @boazjohn is figuring that out at the moment.

**Pull request merged**